plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -342,13 +342,18 @@ maybeAddTySubst tn rhs = do
     else pure $ Just rhs
 
 -- | Is this a an utterly trivial term which might as well be inlined?
-trivialTerm :: Term tyname name uni fun a -> Bool
+trivialTerm
+    :: forall tyname name uni fun a . InliningConstraints tyname name uni fun
+    => Term tyname name uni fun a
+    -> Bool
 trivialTerm = \case
-    Builtin{}  -> True
-    Var{}      -> True
+    Builtin{}                                                                                -> True
+    Var{}                                                                                    -> True
     -- TODO: Should this depend on the size of the constant?
-    Constant{} -> True
-    _          -> False
+    Constant{}                                                                               -> True
+    -- \_ a -> a. Useful to inline then reduce like with `trace` when `remove-trace` is specified.
```

Comment:
I guess from the fact that you put this in that the simplifier _doesn't_ currently inline the "fake trace". But I don't think we should do it like this, this is far too special-cased. Let's think about how to do that separately: for now it's fine if the test output doesn't have it all inlined away, we can worry about that later. 

Basically, we should inline `const`, and we should fix that properly!

Reply:
I agree. I've reverted the `Inine` changes to keep this PR well-scoped for now. Generalizing the simplifier to cover this case (and more!) deserves its own PR.